### DISCUSSION TITLE
smooth curve peaks in DOS plot

### DISCUSSION THREAD

**Discussion by srikasi** [SMTG-UCL/sumo#143] [General]
Hi
while i am ploting the DOS plots, i am getting sharp edged peaks, but not smooth curve/spline like lines. how to get them done like the ones shown in the gallary?

**Comment by ajjackson** [comment]
Broadening can be added using the `-g` argument, which sets the characteristic width of a Gaussian smearing function.

This is not typically necessary for a DOS calculated in VASP with a reasonably large NEDOS and ISMEAR=0 (i.e. Gaussian smearing is applied by VASP). You may have sharp peaks because you used a very fine SIGMA or tetrahedron method (ISMEAR=-5) - in which case it is fine to use the `-g` option in Sumo to add smearing. If NEDOS is too small the plot will look rather "polygonal" and it should be increased before broadening is investigated.

Basically, we try not to smear the data twice so additional broadening is off by default :sweat_smile:

**Comment by srikasi** [comment]
Thank you Adam.

Another issue is that when i created a color code file as below, the colors
are ignored but gave some default colors..
is there any problem with color codes? i used the codes from
this website

https://matplotlib.org/stable/tutorials/colors/colors.html

--- --
[Si]
p = #C20078

[P]
s = #15B01A
p = #0343DF

[As]
s = b
p = r
d = g
----- ----
Regards

Sri

**Comment by srikasi** [comment]
hi SMTG-UCL/sumo

kindly reply regarding the colo config file not reading, not error as well.
also, if i want color at elemental level itself and not to the orbitals,
then how to assign color?
Regards

Sri

**Comment by ajjackson** [comment]
Those colour codes load for me. Writing them to a file in the current directory called *my_colours.conf*, and running `sumo-dosplot -f ~/src/sumo/tests/data/Cs2SnI6/dos/vasprun.xml.gz --config my_colours.conf --format png` with the latest version of Sumo gives

![dos](https://user-images.githubusercontent.com/2511383/138673091-3f1b3a87-d998-4050-b136-186866c95ddf.png)

which is a different set of colours from running without the `--config` argument. Can you check the command you are using to run sumo-dosplot?

I don't recall if we have a way to collect orbitals into whole-element contributions. Any idea @utf ? It would be nice to make the syntax for specifying different configurations simpler and more powerful. The hard part is doing both at once!

**Comment by srikasi** [comment]
hi
Thank you.
After installing '*sumo*', again. and then the colors worked. Thank you.
for example the following color code file








*[Pb]p = r[Br]s = #15B01Ap = #15B01A[O]p = #BBF90F*

results below, which i checked and are correct.

[image: image.png]

Now, I think I can customise the line thickness (??) and font size and font
type (??) etc. I saw some documentation. Will look into that.
Font type or style shall match with other graphs in the same publication,
so if there is an option to choose the font style also be good.
*Note: *Very good work done by you all, thanks once again, and keep it up
with the work. all the best.

NB: if the color code is available at the whole elemental level, then it
would be very nice as an option. As currently, for example I was comparing
band structure variation with different doping models. at that stage, we
sometimes need just elemental level contribution. That is plotting with
data of individual 'total DOS' without orbital resolution.


Regards

Sri

**Comment by srikasi** [comment]
Hi
do we have any error code/ type FAQ?

when i try to plot sumobandplot
it gives the following messages and no plot was drawn
---------
SKM $ sumo-bandplot
Traceback (most recent call last):
  File "/home/595/skm595/.local/bin/sumo-bandplot", line 8, in <module>
    sys.exit(main())
  File
"/home/595/skm595/.local/lib/python3.9/site-packages/sumo/cli/bandplot.py",
line 796, in main
    bandplot(
  File
"/home/595/skm595/.local/lib/python3.9/site-packages/sumo/cli/bandplot.py",
line 273, in bandplot
    bs = vr.get_band_structure(line_mode=True)
  File
"/home/595/skm595/.local/lib/python3.9/site-packages/pymatgen/io/vasp/outputs.py",
line 921, in get_band_structue
    if "" in kpoint_file.labels:
TypeError: argument of type 'NoneType' is not iterable
---------- -
is this related to some issue in my input file?

Regards

Sri

**Comment by srikasi** [comment]
awaiting your reply please

Regards

Sri

**Comment by ajjackson** [comment]
Hi Sri,

this is an error from Pymatgen, which was unable to read the high-symmetry labels from your KPOINTS file. Is this a KPOINTS file from `sumo-kgen` or somewhere else? Does it include the information about high-symmetry labels?


Specifically,
```
    if "" in kpoint_file.labels:
TypeError: argument of type 'NoneType' is not iterable
```
means that it expected `kpoint_file.labels` to be a set of data that it could check for empty entries; but in fact it was `None`, a Python object for null/unset values.
